app/src/main/java/com/android/unio/model/authentication/AuthViewModel.kt

```diff
@@ -31,6 +32,19 @@ constructor(private val firebaseAuth: FirebaseAuth, private val userRepository:
   val authState: StateFlow<String?>
     get() = _authState.asStateFlow()
 
+  private var _credential: AuthCredential? = null
+  val credential: AuthCredential?
+    get() = _credential
+
+  /**
+   * Sets the [AuthCredential] object to be used for linking accounts.
+   *
+   * @param credential The [AuthCredential] object to set.
+   */
+  fun setCredential(credential: AuthCredential?) {
+    _credential = credential
+  }
+
```

Comment:
Rather than having for _crediential a getter and setter that does not impose any constraints on the credential param, couldn't we have _credential public?